promql/engine.go

```diff
@@ -576,26 +576,57 @@ func (ng *Engine) execEvalStmt(ctx context.Context, query *query, s *parser.Eval
 	return mat, warnings, nil
 }
 
-// subqueryOffsetRange returns the sum of offsets and ranges of all subqueries in the path.
-func (ng *Engine) subqueryOffsetRange(path []parser.Node) (time.Duration, time.Duration) {
+// subqueryOffsetRangeTimestamp returns the sum of offsets and ranges of all subqueries in the path.
+// It also returns the min and max timestamp associated with the subquery.
+func (ng *Engine) subqueryOffsetRangeTimestamp(path []parser.Node) (time.Duration, time.Duration, int64, int64) {
 	var (
-		subqOffset time.Duration
-		subqRange  time.Duration
+		subqOffset, subqRange      time.Duration
+		minTimestamp, maxTimestamp int64 = math.MaxInt64, math.MinInt64
 	)
 	for _, node := range path {
 		switch n := node.(type) {
 		case *parser.SubqueryExpr:
 			subqOffset += n.Offset
```

Comment:
This doesn't feel right, shouldn't all of these settings be interacting here? E.g. a fixed timestamp could undo the effect of an offset.

Reply:
I am resetting the offset for calculation of the offset for the selectors now. For here, resetting is going to be tricky with the nested @ modifiers. So I would propose keeping it like this for simplicity and address it if we face issues (the affect is only that we might end up selecting more samples than needed, no other harm otherwise).